libraries/rush-lib/src/utilities/npmrcUtilities.ts

```diff
@@ -22,27 +22,55 @@ export interface ILogger {
 // create a global _combinedNpmrc for cache purpose
 const _combinedNpmrcMap: Map<string, string> = new Map();
 
-function _trimNpmrcFile(options: {
-  sourceNpmrcPath: string;
-  linesToPrepend?: string[];
-  linesToAppend?: string[];
-}): string {
-  const { sourceNpmrcPath, linesToPrepend, linesToAppend } = options;
+function _trimNpmrcFile(
+  options: Pick<
+    INpmrcTrimOptions,
+    'sourceNpmrcPath' | 'linesToAppend' | 'linesToPrepend' | 'supportEnvVarFallbackSyntax'
+  >
+): string {
+  const { sourceNpmrcPath, linesToPrepend, linesToAppend, supportEnvVarFallbackSyntax } = options;
   const combinedNpmrcFromCache: string | undefined = _combinedNpmrcMap.get(sourceNpmrcPath);
   if (combinedNpmrcFromCache !== undefined) {
     return combinedNpmrcFromCache;
   }
+
   let npmrcFileLines: string[] = [];
   if (linesToPrepend) {
     npmrcFileLines.push(...linesToPrepend);
   }
+
   if (fs.existsSync(sourceNpmrcPath)) {
     npmrcFileLines.push(...fs.readFileSync(sourceNpmrcPath).toString().split('\n'));
   }
+
   if (linesToAppend) {
     npmrcFileLines.push(...linesToAppend);
   }
+
   npmrcFileLines = npmrcFileLines.map((line) => (line || '').trim());
+
+  const resultLines: string[] = trimNpmrcFileLines(npmrcFileLines, process.env, supportEnvVarFallbackSyntax);
+
+  const combinedNpmrc: string = resultLines.join('\n');
+
+  //save the cache
+  _combinedNpmrcMap.set(sourceNpmrcPath, combinedNpmrc);
+
+  return combinedNpmrc;
+}
+
+/**
+ *
+ * @param npmrcFileLines The npmrc file's lines
+ * @param env The environment variables object
+ * @param supportEnvVarFallbackSyntax Whether to support fallback values in the form of `${VAR_NAME:-fallback}`
+ * @returns
+ */
+export function trimNpmrcFileLines(
+  npmrcFileLines: string[],
+  env: NodeJS.ProcessEnv,
```

Comment:
If you're using the `Pick<...>` syntax above, may as well do it here too.

Reply:
This isn't an object.